### PR TITLE
Rails forms project

### PR DESCRIPTION
**Pull request by igorkol91**
We made a model for users.
Created a controller with methods new, create, edit, update.
Created a view for new users and edit users.

### CODE REVIEW

File: app/views/users/new.html.erb
Comment (BrianSammit) [inline app/views/users/new.html.erb]: 1. From line 28 to line 37 must be commented out. 
2. In line 28. Instead of `action="/users"` use `action="users_path"` helper for action

File: app/views/users/new.html.erb
Comment (BrianSammit) [inline app/views/users/new.html.erb]: 1. Instead of `/users` use `users_path`
2. In line 15 instead of `text_field` use `text_field_tag`
3. In line 19 instead of `text_field` use `password_field_tag` 
4. In line 23 instead of `text_field` use `email_field_tag`

File: app/views/users/new.html.erb
Comment (BrianSammit) [inline app/views/users/new.html.erb]: 1. In line 39 change this line to `form_for @user do |form|` to make use of `@user`. with `form_with` you will have an error.
2. In line 43 instead of `form.text_field` use `form.password_field`
2. In line 45 instead of `form.text_field` use `form.email_field`

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: :stop_sign: 
-------------
Hello @safafa @igorkol91 great job you made on this project :clap: 
TO HIGHLIGHT: 

- Your project is professional. 
- Your repo is professional
- Well use of the MVP 
- Well structured files

However, there are a few changes to make to complete this project :hammer_and_wrench: see them below :eyes: :point_down: 
-----------

- [ ] On your `users_controller` in update action it redirect to `edit_user_path`. Please fix this.
- [x] Please check the `new.html.erb. See my inline comments :point_down: 
- [x] Please make the same changes from above for the edit form. 
- [x] Please change the name of the readme file. You have `Micro Reddit`
- [x] Please write the steps on getting started more clear. Use bullet points for that, also you can use backticks ` `` ` to highlight the commands that they need to run, for example, `bundle install`. Remember that not every person that gonna see your project knows about programming so please write the steps thinking of someone that doesn't know anything about programming. 
- [ ] [OPTIONAL] Instead of creating twice the form in edit and in new use partials.
STEPS 
1. Create in your views a file called `_form.html.erb`
2. Copy and paste the forms you already create inside `_form.html.erb`
3. Inside the `new.html.erb` and `edit.html.rb` write this line of code. `<%= render partial: 'form' %>`

-----------------
Please make these required changes and send it again for another review :100: :+1: 
Reach me on SLACK if you have any questions. 
@ Brian Sammit Cruz Rodriguez
HAPPY CODING!!

**Comment by igorkol91** [comment]
https://github.com/safafa/re-former/pull/1#discussion_r644115774
Unfortunately, we're unable to change line 39  since in the Odin requirements were using the form_with and we found it to work as expected.
Kindly review.

File: .github/workflows/linter.yml
Comment (carlos-ssh) [inline .github/workflows/linter.yml]: - Missing Setup for Stylelint

File: app/views/users/new.html.erb
Comment (carlos-ssh) [inline app/views/users/new.html.erb:37]: - Kindly avoid commented code.

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hello team! (@safafa and @igorkol91) 👋🏽👋🏽👋🏽
Your project it's very nice!.

Some changes are needed before to approve your project.
Please work on these changes:
- When i go to ``` http://127.0.0.1:3000/users/new ``` I have this error message:
Screen Shot 2021-06-03 at 05 39 27

- Setup files for Stylelint and Rubocop. [Set up a code linter](https://github.com/microverseinc/linters-config/tree/master/ror)
- Please follow the changes requested on the comments below.

Thank you guys, keep working very well.

Happy Coding! 🥷🏽👾
@Carlos-SSH

**Comment by igorkol91** [comment]
We have some commented code on new.html.erb.
It's the forms that we had to make before we needed to use the form_with tag.
It was required in the lesson that we keep them but commented out.
Also https://github.com/safafa/re-former/pull/1#pullrequestreview-675117001 the previous CR had a problem with this line but we managed to solve it just by using the $yarn command in the console. I would guess that he probably had some missing packages. 
Kindly review.

**Review by umararshadbut — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @safafa and @igorkol91 
Good job with the changes! :tada: We are almost there.
We, however, do have a few changes to carry out before we proceed.

##

Kindly work on the suggested changes  👇.
- [x]  Please write the steps on getting started more clear. Use bullet points for that, also you can use backticks `` to highlight the commands that they need to run, for example, `bundle install`. Remember that not every person that gonna see your project knows about programming so please write the steps thinking of someone that doesn't know anything about programming. (As first Code Reviewer suggeded)


##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you are stuck at any point or have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**Umair**

File: README.md
Comment (enaburekhan) [inline README.md]: As requested by the last reviewer. You need to write detailed and complete instructions required to run the app locally. The instructions should be clear enough even for nonprogrammers.  Let me continue from where you stop.
enter `http://localhost:3000/users/new` to create new users
enter `http://localhost:3000/users/1/edit` to edit existing users

File: README.md
Comment (enaburekhan) [inline README.md:31]: enter `User.all` to confirm the created user

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi team!
Great job so far:clap::clap:. We are almost there. Just this final change and this project will be approved:muscle:.**

**To improve**

- [ ] Check my inline comments in the readme and make the required changes.

Request another review when you are ready.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Review by umararshadbut — APPROVED** [review]
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @igorkol91  and @safafa ! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
HAPPY CODING :computer:
**Umair**

**Pull request merged**
